src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseSign1Message.cs

```diff
@@ -209,39 +209,57 @@ private static ReadOnlyMemory<byte> GetCoseAlgorithmFromProtectedHeaders(CoseHea
             // This authentication can be done either by placing the header in the protected header bucket or as part of the externally supplied data.
             if (!protectedHeaders.TryGetEncodedValue(CoseHeaderLabel.Algorithm, out ReadOnlyMemory<byte> encodedAlg))
             {
-                throw new CryptographicException(SR.Sign1SignAlgIsRequired);
+                throw new CryptographicException(SR.Sign1VerifyAlgIsRequired);
             }
 
             return encodedAlg;
         }
 
         // If we Validate: The caller did specify a COSE Algorithm, we will make sure it matches the specified key and hash algorithm.
-        // If we Add: The caller did not specify a COSE Algorithm, we will add the header for them, rather than throw. protectedHeaders will be altered in this case.
-        private static void ValidateOrAddAlgorithmHeader(CoseHeaderMap protectedHeaders, CoseHeaderMap unprotectedHeaders, KeyType keyType, HashAlgorithmName hashAlgorithm)
+        // If we Slip: The caller did not specify a COSE Algorithm, we will write the header for them, rather than throw.
+        private static void ValidateOrSlipAlgorithmHeader(
+            CoseHeaderMap protectedHeaders,
+            CoseHeaderMap unprotectedHeaders,
+            KeyType keyType,
+            HashAlgorithmName hashAlgorithm,
+            out int? headerValueToSlip)
```

Comment:
`void` return with a single `out` seems strange.  Why aren't we just returning it?

Reply:
I thought an out parameter was nicer since it allows me to name the return value (which in this case I think is not obvious).